ledger/participant-integration-api/src/main/scala/platform/apiserver/ApiServerConfig.scala

```diff
@@ -4,31 +4,61 @@
 package com.daml.platform.apiserver
 
 import com.daml.ledger.api.tls.TlsConfiguration
-import com.daml.lf.data.Ref
 import com.daml.platform.apiserver.SeedService.Seeding
-import com.daml.platform.configuration.{IndexConfiguration, InitialLedgerConfiguration}
-import com.daml.ports.Port
-import java.nio.file.Path
-import java.time.Duration
-
+import com.daml.platform.configuration.{
+  CommandConfiguration,
+  InitialLedgerConfiguration,
+  PartyConfiguration,
+}
+import com.daml.platform.services.time.TimeProviderType
+import com.daml.platform.store.DbSupport.{ConnectionPoolConfig, DbConfig}
 import com.daml.platform.usermanagement.UserManagementConfig
+import com.daml.ports.Port
 
-import scala.concurrent.duration.FiniteDuration
+import java.nio.file.Path
+import scala.concurrent.duration._
 
 case class ApiServerConfig(
-    participantId: Ref.ParticipantId,
-    port: Port,
-    address: Option[String], // This defaults to "localhost" when set to `None`.
-    jdbcUrl: String,
-    databaseConnectionPoolSize: Int,
-    databaseConnectionTimeout: FiniteDuration,
-    tlsConfig: Option[TlsConfiguration],
-    maxInboundMessageSize: Int,
-    initialLedgerConfiguration: Option[InitialLedgerConfiguration],
-    configurationLoadTimeout: Duration,
-    indexConfiguration: IndexConfiguration,
-    portFile: Option[Path],
-    seeding: Seeding,
-    managementServiceTimeout: Duration,
-    userManagementConfig: UserManagementConfig,
+    port: Port = ApiServerConfig.DefaultPort,
+    address: Option[String] =
+      ApiServerConfig.DefaultAddress, // This defaults to "localhost" when set to `None`.
+    tls: Option[TlsConfiguration] = ApiServerConfig.DefaultTls,
+    maxInboundMessageSize: Int = ApiServerConfig.DefaultMaxInboundMessageSize,
+    initialLedgerConfiguration: Option[InitialLedgerConfiguration] =
+      ApiServerConfig.DefaultInitialLedgerConfiguration,
+    configurationLoadTimeout: Duration = ApiServerConfig.DefaultConfigurationLoadTimeout,
+    portFile: Option[Path] = ApiServerConfig.DefaultPortFile,
+    seeding: Seeding = ApiServerConfig.DefaultSeeding,
+    managementServiceTimeout: FiniteDuration = ApiServerConfig.DefaultManagementServiceTimeout,
+    userManagement: UserManagementConfig = ApiServerConfig.DefaultUserManagement,
+    authentication: AuthServiceConfig = ApiServerConfig.DefaultAuthentication,
+    party: PartyConfiguration = ApiServerConfig.DefaultParty,
+    command: CommandConfiguration = ApiServerConfig.DefaultCommand,
+    timeProviderType: TimeProviderType = ApiServerConfig.DefaultTimeProviderType,
+    database: DbConfig = ApiServerConfig.DefaultDatabase,
 )
+
+object ApiServerConfig {
+  val DefaultPort: Port = Port(6865)
```

Comment:
An alternative system would have been to inline the value defaults into the class and then have `val Default = ApiServerConfig()`

Reply:
I had them extracted because some of the cases were exposed to LegacyCLI. I will change it if no field is exposed somewhere else.